tests/testthat/test-src.R

```diff
@@ -0,0 +1,144 @@
+test_that("Can migrate orderly demo directory", {
+  path <- orderly_demo_src()
+  orderly2outpack_src(path, delete_yml = TRUE, strict = TRUE)
+
+  dat <- orderly:::read_demo_yml(path)
```

Comment:
I think this is not testing every report can be run? In particular `default-param` is not being run

```
> reports <- vapply(dat, function(x) x$name, character(1))
> reports
 [1] "minimal"            "minimal"            "other"              "minimal"            "other"              "global"            
 [7] "use_dependency"     "other"              "other"              "changelog"          "use_dependency"     "use_resource"      
[13] "multi-artefact"     "multifile-artefact" "changelog"          "html"               "interactive"        "use_resource_dir"  
[19] "connection"         "spaces"             "view"     
```

Reply:
yeah, this only runs the cases that are in the demo